partner_firstname/models/res_partner.py

```diff
@@ -31,8 +31,14 @@ def create(self, vals):
         """Add inverted names at creation if unavailable."""
         context = dict(self.env.context)
         name = vals.get("name", context.get("default_name"))
-
-        if name is not None:
+        if context.get("copy") and vals.get("firstname") and not vals.get("is_company"):
+            vals["firstname"] = _("%s (copy)", vals["firstname"])
+            # Remove the combined fields
+            if "name" in vals:
+                del vals["name"]
+            if "default_name" in context:
```

Comment:
You should instead override the `copy` function directly.

Reply:
Thanks for feeeback @Yajo 
Done!

Regards!